doc/man3/ASYNC_WAIT_CTX_new.pod

```diff
@@ -103,14 +120,59 @@ code that the job should be resumed by making the wait file descriptor
 "readable". Once resumed the engine should clear the wake signal on the wait
 file descriptor.
 
+As well as file descriptor, user code may also be notified via a callback. The
+callback and data pointers would be stored within the ASYNC_WAIT_CTX along with
+an additional status field that can be used for the notification of retries
+from an engine. This additional method can be used when user thinks the file
+descriptor is too costly in terms of CPU cycles or in some context where file
+descriptor is not appropriate.
+
+Below APIs are defined for this new notification way.
+
+ASYNC_WAIT_CTX_set_callback() sets the callback and the callback argument. The
+callback will be called to notify user code when an engine completes a
+cryptography operation.
+
+ASYNC_WAIT_CTX_get_callback() returns the callback set in ASYNC_WAIT_CTX
+strucutre. It is normally used for an engine to check if this new notification
+way is enabled.
+
+ASYNC_WAIT_CTX_set_status() allows an engine to set current engine status, so
+that user code can understand the engine cryptography status. The possible
+status values are the following:
+ASYNC_STATUS_UNSUPPORTED: Callback mechanism does not supported by this engine.
+ASYNC_STATUS_ERR: HW has fatal problem, thus the crypto request is not submitted
+to HW, and suggest user code to stop this session.
+ASYNC_STATUS_OK: Everything is fine, and the request is successfully sent to HW.
+ASYNC_STATUS_EAGAIN: The engine has some problem which will be recovered soon,
+such as buffer is full, so user code should send the request again next time.
```

Comment:
In my code implementing in VPP:
firstly I will create a queue to put the session related into into an event queue.
then I will call below to put the event id as argument.
_SSL_set_async_callback_arg (oc->ssl, (void *) engine_cb->arg);_
third: call SSL_do_handshake
fourth: check the return value
In this case: if "ASYNC_STATUS_EAGAIN", then it means,  the request has NOT been sent, but there is no chance to call the callback, so user code here will call the callback, the callback will put event id to running status, and afterwards resume the  SSL_do_handshake and use fiber to "re-submit" the request. 
If SYNC_STATUS_OK, then there is no retry logic, and it continue for next session. 
If YNC_STATUS_ERR, then it means the engine has some problem. It will help call the callback to put the session to running status, and afterwards, it will not call SSL_do_handshake any more, but just cleanup the session.